gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,238 @@
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+    """
+
+    _connection_class = Connection
+
+    def list_projects(self, filter_params=None, page_size=None):
+        """List the projects visible to this client.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> for project in client.list_projects():
+            ...     print project.project_id
+
+        List all projects with label ``'environment'`` set to ``'prod'``
+        (filtering by labels)::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> env_filter = {'labels.environment': 'prod'}
+            >>> for project in client.list_projects(env_filter):
+            ...     print project.project_id
+
+        See:
+        https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/list
+
+        Complete filtering example::
+
+            >>> project_filter = {  # Return projects with...
+            ...     'name': 'My Project',  # name set to 'My Project'.
+            ...     'id': 'my-project-id',  # id set to 'my-project-id'.
+            ...     'labels.stage': 'prod',  # the label 'stage' set to 'prod'
+            ...                           # set to prod.
+            ...     'labels.color': '*'  # a label 'color' set to anything.
+            ... }
+            >>> client.list_projects(project_filter)
+
+        :type filter_params: dict
+        :param filter_params: A dictionary of filter options where the keys are
+                              the property to filter on, and the value is the
+                              case-insensitive value to check (or * to check
+                              for existence of the property). See the example
+                              above for more details.
+                              Note that property values are case-insensitive.
+
+        :type page_size: int
+        :param page_size: maximum number of projects to return in a single
+                          page. If not passed, defaults to a value set by the
+                          API.
+
+        :rtype: :class:`gcloud.resource_manager.iterator.ProjectIterator`
+        :returns: A ProjectIterator class, which allows you to iterate through
+                  all of the results without thinking about pagination.
+                  Each item will be a :class:`.project.Project` object.
+        """
+        params = {}
+
+        if page_size is not None:
+            params['pageSize'] = page_size
+
+        if filter_params is not None:
+            params['filter'] = filter_params
+
+        client = self
+
+        class ProjectIterator(Iterator):
```

Comment:
Please define this at the global level. I'd also prefer it were private (i.e `_ProjectIterator`).